src/raster/r.suitability.regions/r.suitability.regions.py

```diff
@@ -0,0 +1,410 @@
+#% guisection: Input
+#%end
+#%option
+#% key: minsuit
+#% description: Threshold value above which areas are considered suitable
+#% required: yes
+#% type: string
+#% answer: 0.7
+#% key_desc: float
+#% guisection: Input
+#%end
+#%option
+#% key: minfrag
+#% description: Minimum area (in hectares) limit for fragments
+#% required: yes
+#% type: string
+#% answer: 1000
+#% key_desc: float
+#% guisection: Input
+#%end
+#%option
+#% key: absminsuit
+#% description: Threshold below which cells are marked as fully unsuitable
+#% required: no
+#% type: string
+#% key_desc: float
+#% guisection: Input
+#%end
+
+#%flag
+#% key: d
+#% label: Clumps including diagonal neighbors
+#% description: Clumps including diagonal neighbors
+#% guisection: Optional
+#%end
+
+#%option
+#% key: radius
+#% description: Radius for focal stat
+#% required: no
+#% type: integer
+#% answer: 1
+#% guisection: Focal stats
+#%end
+#%flag
+#% key: c
+#% label: Circular neighborhood
+#% description: Use circular neighborhood for focal statistic
+#% guisection: Focal stats
+#%end
+#%option
+#% key: focalstat
+#% description: Neighborhood operation (focal statistic)
+#% required: no
+#% type: string
+#% answer: median
+#% options:maximum,median,quart1,quart3
+#% guisection: Focal stats
+#%end
+
+#%option
+#% key: maxgap
+#% description: Maximum gap size (in hectares) to remove
+#% required: no
+#% type: string
+#% answer: 0
+#% key_desc: float
+#% guisection: Remove gaps
+#%end
+
+#%flag
+#% key: z
+#% label: Suitability per region
+#% description: Compute map with average suitability per region
+#% guisection: Reporting stats
+#%end
+#%flag
+#% key: a
+#% label: Calculates area of clumped areas (hectares)
+#% description: Calculates area of clumped areas
+#% guisection: Reporting stats
+#%end
+
+#%flag
+#% key: k
+#% label: Keep map with all suitable cells
+#% description: Keep map that show suitable areas (irrespective of clump size)
+#% guisection: Optional
+#%end
+
+#%flag
+#% key: f
+#% label: Keep suitability based on focal statistics
+#% description: Keep suitabiliyt based on focal statistics
+#% guisection: Optional
+#%end
+
+import sys
+import os
+import atexit
+import uuid
+import grass.script as gs
+
+RECLASSRULES1 = """\
+1:filled gaps
+2:suitable areas
+"""
+COLORRULES1 = """
+1 230:230:230
+""".strip()
+
+COLORRULES2 = """
+1 241:241:114
+2 139:205:85
+""".strip()
+
+
+CLEAN_LAY = []
+
+
+def cleanup():
+    """Remove temporary maps specified in the global list"""
+    cleanrast = list(reversed(CLEAN_LAY))
+    for rast in cleanrast:
+        ffile = gs.find_file(name=rast, element="cell", mapset=gs.gisenv()["MAPSET"])
+        if ffile["file"]:
+            gs.run_command("g.remove", flags="f", type="raster", name=rast, quiet=True)
+
+
+def tmpname(prefix):
+    """Generate a tmp name which contains prefix
+    Store the name in the global list.
+    Use only for raster maps.
+    """
+    tmpf = prefix + str(uuid.uuid4())
+    tmpf = tmpf.replace("-", "_")
```

Comment:
See new functions in gs (`grass/script/utils.py`) for this. If you look for v7 compatibility, import in `try` and use copy-pasted (possibly simplified) version in except. See e.g.:

https://github.com/OSGeo/grass-addons/blob/grass7/src/misc/m.csv.clean/m.csv.clean.py#L90

Reply:
Useful new function. In this particular case, the easier (less code) solution I found is to use str(uuid.uuid4().hex) which removes the hypen from the name in one go.

Reply:
The particular function is not that relevant here. I meant the link as an example of using import with try-except for version compatibility.

The hex trick is implemented in `gs.append_uuid` (`python/grass/script/utils.py`) together with possibly a better alternative `append_node_pid`. Using one or the other provides an opportunity for future improvement (at one single place in the library) or for correction where a clearly identifiable function can be detected in modules, so it makes sense to use a function even in place of the short code.

https://github.com/OSGeo/grass/blob/master/python/grass/script/utils.py#L537